### PR TITLE
[DOC] Enhanced RDoc for Time

### PR DESCRIPTION
**Pull request by BurdetteLamar**
More on timezones.

### CODE REVIEW

File: doc/timezones.rdoc [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+In most cases, the zone value may be an instance
+of a specialized timezone class.
+
+\Exception (timezone object not allowed):
+
+- Time.new with positional argument +zone+.
+- Time.now with keyword argument +in:+.
+
+===== +TZInfo::Timezone+ Subclasses
```
Comment (jeremyevans) [inline doc/timezones.rdoc:68]: Remove this section.  The documentation should describe the API required for timezone objects, it should not provide details of which gems implement compatible objects.

File: doc/timezones.rdoc [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  tz = TZInfo::Timezone.get('America/New_York')
+  # => #<TZInfo::DataTimezone: America/New_York>
+  tz.kind_of?(TZInfo::Timezone) # => true
+
+===== Custom Timezone Classes
+
+A custom timezone object +tz+ may be used as the value for +zone+
+in creating a new \Time object +t+.
+
+A custom timezone class must have these instance methods:
```
Comment (jeremyevans) [inline doc/timezones.rdoc:98]: `s/A custom timezone class/The timezone object/`

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Please see inline comments.

File: timev.rb [hunk 3]
```diff
@@ -208,38 +208,6 @@
 # - #ceil: Returns a new time with subseconds raised to a ceiling.
 # - #floor: Returns a new time with subseconds lowered to a floor.
 #
-# == Timezone Argument
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply timev.rb:211]: Done.

File: doc/timezones.rdoc [hunk 4]
```diff
@@ -0,0 +1,139 @@
+== Timezones
+
+=== Timezone Specifiers
+
+Certain Ruby methods accept arguments that specify timezones:
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/timezones.rdoc]: Done.

File: doc/timezones.rdoc [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+The zone value may be an integer number of seconds
+in the range <tt>-86399..86399</tt>:
+
+  t = Time.utc(2000, 1, 1, 20, 15, 1) # => 2000-01-01 20:15:01 UTC
+  Time.at(t, in: -86399)              # => 1999-12-31 20:15:02 -235959
+  Time.at(t, in: 86399)               # => 2000-01-02 20:15:00 +235959
+
+==== Timezone Objects
+
+In most cases, the zone value may be an instance
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/timezones.rdoc]: Done.

File: doc/timezones.rdoc (hunk 1 above)
Comment (BurdetteLamar) [reply doc/timezones.rdoc]: Done.

File: doc/timezones.rdoc (hunk 2 above)
Comment (BurdetteLamar) [reply doc/timezones.rdoc]: Done.

**Review by jeremyevans — APPROVED** [review]

File: doc/timezones.rdoc [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    and when Time#getlocal or Time#localtime is called with +tz+
+    as the value for positional argument +zone+.
+  - Argument: a <tt>Time::tm</tt> object.
+  - Returns: whether the time is daylight saving time.
+
+- +name+:
+
+  - Called when <tt>Marshal.dump(t) is invoked
+  - Argument: none.
+  - Returns: the string name of the time.
```
Comment (nobu) [inline doc/timezones.rdoc:108]: Maybe?
```suggestion
  - Returns: the string name of the timezone.
```

**Pull request merged**